### PR TITLE
feat(tools): add color palette to tailwind config file

### LINKED ISSUE #42186: Add color tokens to tailwind config file
We are in the progress of building a component library, and [TailwindCSS](https://tailwindcss.com/docs) is what we use for component theming.

The config file has been created, and further configurations will be added in the future. For now, we need to add color tokens to the file, in the `colors` object. 

Link to the config file: https://github.com/freeCodeCamp/freeCodeCamp/blob/main/tools/ui-components/tailwind.config.js#L5

Colors to be added:
- Layout colors: https://design-style-guide.freecodecamp.org/#layout-colors
- Accent colors: https://design-style-guide.freecodecamp.org/#accent-colors
- Any colors in here that the design style guide does not have: https://github.com/freeCodeCamp/freeCodeCamp/blob/main/client/src/components/layouts/variables.css

Convention:
- Token names are in camelCase
- Tokens for dark theme colors are suffixed with the word `dark`. Tokens for light theme can stay as is. For example: `primaryColor` and `primaryColorDark`, `secondaryColor` and `secondaryColorDark`, etc.
- Tokens are sorted alphabetically

### PR DESCRIPTION
**Pull request by ChrisPrzR**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Ref #42186

### CODE REVIEW

**Comment by gikf** [comment]
Hey @ChrisPrzR,

We would love to be able to merge your changes but it looks like there are an errors with the style linting reported by CI checks. ⚠️

Once you resolve these issues, we will be able to review your PR and merge it. 😊

File: tools/ui-components/tailwind.config.js [hunk 1]
```diff
@@ -4,7 +4,39 @@ module.exports = {
...
+      gray00: '#ffffff',
+      gray05: '#f5f6f7',
+      gray10: '#dfdfe2',
+      gray15: '#d0d0d5',
+      gray45: '#858591',
+      gray75: '#3b3b4f',
+      gray80: '#2a2a40',
+      gray85: '#1b1b32',
+      gray90: '#0a0a23',
+      //Accent Colors - Primary
```
Comment (nhcarrigan) [inline tools/ui-components/tailwind.config.js:17]:
```suggestion
      // Accent Colors - Primary
```

File: tools/ui-components/tailwind.config.js [hunk 2]
```diff
@@ -4,7 +4,39 @@ module.exports = {
...
+      blue: '#99c9ff',
+      green: '#acd157',
+      purple: '#dbb8ff',
+      yellow: '#f1be32',
+      //Accent Colors - Secondary
```
Comment (nhcarrigan) [inline tools/ui-components/tailwind.config.js:22]:
```suggestion
      // Accent Colors - Secondary
```

File: tools/ui-components/tailwind.config.js [hunk 3]
```diff
@@ -4,7 +4,39 @@ module.exports = {
...
+      darkBlue: '#002ead',
+      darkGreen: '#00471b',
+      darkPurple: '#5a01a7',
+      darkYellow: '#4d3800',
+      //Colors in variables.css
```
Comment (nhcarrigan) [inline tools/ui-components/tailwind.config.js:27]:
```suggestion
      // Colors in variables.css
```

File: tools/ui-components/tailwind.config.js [hunk 4]
```diff
@@ -4,7 +4,39 @@ module.exports = {
...
+      blueMid: '#198eee',
+      blueTranslucent: 'rgba(153, 201, 255, 0.3)',
+      darkBlueTranslucent: 'rgba(0, 46, 173, 0.3)',
+      darkEditorBackground: '#2a2b40',
+      darkLove: '#f82153',
+      darkRed: '#850000',
+      editorBackground: '#fffffe',
+      love: '#f8577c',
+      purpleMid: '#9400d3',
+      red: '#ffadad',
```
Comment (nhcarrigan) [inline tools/ui-components/tailwind.config.js:37]:
```suggestion
      red: '#ffadad',
```

File: tools/ui-components/tailwind.config.js [hunk 5]
```diff
@@ -4,7 +4,39 @@ module.exports = {
...
+      darkBlueTranslucent: 'rgba(0, 46, 173, 0.3)',
+      darkEditorBackground: '#2a2b40',
+      darkLove: '#f82153',
+      darkRed: '#850000',
+      editorBackground: '#fffffe',
+      love: '#f8577c',
+      purpleMid: '#9400d3',
+      red: '#ffadad', 
+      yellowGold: '#ffbf00',
+      yellowLight: '#ffc300',
```
Comment (nhcarrigan) [inline tools/ui-components/tailwind.config.js:39]:
```suggestion
      yellowLight: '#ffc300'
```

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
Hey there @ChrisPrzR 

These suggestions should resolve the linting issues.

**Comment by ChrisPrzR** [comment]
@gikf @nhcarrigan  Thank you! I've applied the suggested changes .

**Review by huyenltnguyen — CHANGES_REQUESTED** [review]
@ChrisPrzR Thank you for this PR and for the fix in variables.css ealier!

There are a couple more changes we need to make, and the PR will be good to go then.

As you can see in this file, we are having quite a lot of colors with the name "dark", but some of them are actually dark, as in shade; while the other refer to the dark theme. We will need to work on a new set of color tokens (partially discussed in https://github.com/freeCodeCamp/freeCodeCamp/issues/42186#issuecomment-845829904), but to avoid confusion for now, I'd suggest:
- If the color is dark in shade, we name it `dark[Color]`
- If the color is used in dark theme, we name it `[color]Dark`

**Comment by ChrisPrzR** [comment]
@huyenltnguyen these changes look great, I like the readability improvement. Would you suggest leaving dark colors used in both themes as dark[Color] ?

**Comment by huyenltnguyen** [comment]
Yes, leaving them `dark[Color]` would be fine, although I could imagine it would cause confusion for other readers still, with all the "dark" suffixes and prefixes.

But don't worry about it. We will change them soon as we add the dark theme support to our component library 🙂

**Review by huyenltnguyen — APPROVED** [review]
LGTM 👍 

Thank you again for the PR and we are looking forward to your future contributions.

Stay safe and happy coding!

**Review by naomi-lgbt — APPROVED** [review]

**Comment by nhcarrigan** [comment]
Before I merge, @huyenltnguyen and @ahmadabdolsaheb can you confirm this does resolve the linked issue?

**Comment by huyenltnguyen** [comment]
Thanks for checking @nhcarrigan! 

Some idea came up after we opened the issue up for contributions, and the puzzle has more pieces now. This PR adds one in, so I have replaced the word "closes" to "ref" in the PR description to keep the issue around. I will go back and update the label of the issue to "discussing".

**Pull request merged**